api/pkg/db/migration/202109201733_add-code_col_in_user_table.go

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+func addCodeColInUserTable(log *log.Logger) *gormigrate.Migration {
+
+	return &gormigrate.Migration{
+		ID: "202109201733_create_user_table",
+		Migrate: func(db *gorm.DB) error {
+			if err := db.Migrator().AddColumn(&model.User{}, "code"); err != nil {
```

Comment:
prev we used to do `AutoMirgrate()` after adding new col name in struct
here we are doing AddColumn, any diff between the two?

Reply:
AutoMigrate will create tables, missing foreign keys, constraints, columns and indexes. It will change existing column’s type if its size, precision, nullable changed while add column will just add the column, so thought this might be more optimized one hence adding a column

Reply:
is it `optimized`? under the hood are the query different?
i don't mind keep any of them just asking !

Reply:
I mean optimizing in the sense that we don't need to create the whole table just add a column

Reply:
Automigrate is good when there are multiple changes, here as we are just adding a column with no reference anywhere like foreign key etc, it is fine to use Add Column